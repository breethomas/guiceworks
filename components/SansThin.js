// @flow
import React from 'react'
import css from '../styles/css'

css.fontFace({
  fontFamily: 'Helvetica-Th',
  fontStyle: 'normal',
  fontWeight: 400,
  src: 'url("/static/fonts/HelveticaNeueLTStd-Th.otf") format("opentype")',
})

const style = css({
  fontFamily: 'Helvetica-Th, "Helvetica Neue", Helvetica, sans-serif',
  fontWeight: 200,
})

type Props = {
  className?: string,
  tag?: string,
}

const SansThin = ({ className, tag, ...props }: Props) => {
  const Component = tag != null ? tag : 'span'
  return (
    <Component className={`${style} ${className || ''}`} {...props} />
  )
}

SansThin.defaultProps = {
  className: '',
  tag: 'span',
}

export default SansThin
