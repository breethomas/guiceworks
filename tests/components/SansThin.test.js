import React from 'react'
import renderer from 'react-test-renderer'
import SansThin from '../../components/SansThin'

describe('SansThin', () => {
  let component

  beforeEach(() => {
    component = (
      <SansThin className="SansThinClassName" tag="p">
        SansThin
      </SansThin>
    )
  })

  it('shows the correct snapshot for the SansThin component tree', () => {
    const tree = renderer.create(component).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
