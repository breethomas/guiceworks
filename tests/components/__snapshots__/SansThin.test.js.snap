// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SansThin shows the correct snapshot for the SansThin component tree 1`] = `
<p
  className="css-isubv6 SansThinClassName"
>
  SansThin
</p>
`;
